### PR TITLE
add libdavs2 decoder v2 -hwrenx

### PR DESCRIPTION
**Pull request by hwrenx**
Add avs2 video decoder.

### CODE REVIEW

File: libavcodec/libdavs2.c [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+#include "avcodec.h"
+#include "libavutil/imgutils.h"
+#include "internal.h"
+
+#include <davs2.h>
+
+typedef struct DAVS2Context {
+    AVCodecContext *avctx;
+
+    void *decoder;
```
Comment (T-bagwell) [inline libavcodec/libdavs2.c:37]: Why does this is void? maybe the davs2_decoder_open should return a davs_config or other paranter struct?

File: libavcodec/libdavs2.c [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+    davs2_packet_t   packet;     // input bitstream
+
+    int decoded_frames;
+
+    davs2_picture_t  out_frame;  // output data, frame data
+    davs2_seq_info_t headerset;  // output data, sequence header
+
+}DAVS2Context;
+
+static av_cold ff_davs2_init(AVCodecContext *avctx)
```
Comment (T-bagwell) [inline libavcodec/libdavs2.c:50]: reference to ：https://ffmpeg.org/developer.html#Naming-conventions
For variables and functions visible outside of file scope, but only used internally by a library, an ff_ prefix should be used, e.g. ‘ff_w64_demuxer’.


maybe remove the ff_ from here is ok.

File: libavcodec/libdavs2.c [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    frame->height    = cad->headerset.vertical_size;
+    frame->pts       = cad->out_frame.pts;
+    frame->pict_type = pic->type;
+    frame->format    = avctx->pix_fmt;
+
+    cad->decoded_frames++;
+    return 1;
+}
+
+static av_cold ff_davs2_end(AVCodecContext *avctx)
```
Comment (T-bagwell) [inline libavcodec/libdavs2.c:107]: rename ff_davs2_end to davs2_end

File: libavcodec/libdavs2.c [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+
+static int davs_dump_frames(AVCodecContext *avctx, davs2_picture_t *pic, davs2_seq_info_t *headerset, AVFrame *frame)
+{
+    DAVS2Context *cad = avctx->priv_data;
+    avctx->flags |= AV_CODEC_FLAG_TRUNCATED;
+    int bytes_per_sample = pic->bytes_per_sample;
+    int i;
+
+    if (!headerset)
+        return AVERROR(EINVAL);
```
Comment (T-bagwell) [inline libavcodec/libdavs2.c:72]: if return AVERROR(EINVAL); here, the return check got_frame maybe >=0 is better, maybe this logic need double check.

File: libavcodec/libdavs2.c [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+    cad->frame = frame;
+    avctx->flags |= AV_CODEC_FLAG_TRUNCATED;
+
+    if (!buf_size) {
+        cad->packet.data = buf_ptr;
+        cad->packet.len  = buf_size;
+        cad->packet.pts  = avpkt->pts;
+        cad->packet.dts  = avpkt->dts;
+
+        while (true) {
```
Comment (T-bagwell) [inline libavcodec/libdavs2.c:139]: what about rename true to 1

File: libavcodec/libdavs2.c (hunk 1 above)
Comment (hwrenx) [reply libavcodec/libdavs2.c:37]: just a pointer...

File: libavcodec/libdavs2.c (hunk 4 above)
Comment (hwrenx) [reply libavcodec/libdavs2.c:72]: return 0 here and check if (!*got_frame) may work well.

File: libavcodec/libdavs2.c (hunk 1 above)
Comment (T-bagwell) [reply libavcodec/libdavs2.c:37]: why not davs2_mgr_t?

**Pull request closed without merging**